{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### The approach here is to distinguish the variants in multi-variant case. \n",
    "1. Use my defination to order the variant type\n",
    "2. Find the coresponding info such as AF, allele sequencing, allele_conut, etc.\n",
    "3. Use a datafame to store the informaiton and to munipulate\n",
    "4. Use \"requests' to query external API-ExAC for population informaiton such as allele frequency."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import all the package\n",
    "import vcf\n",
    "import allel\n",
    "import h5py\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import requests\n",
    "import pprint \n",
    "pd.set_option('display.max_columns', None)\n",
    "\n",
    "#Load the file\n",
    "filename = 'challenge_data.vcf'\n",
    "#Save it to a npz file ehich is easier for me to process\n",
    "allel.vcf_to_npz(filename, 'example.npz', fields='*', overwrite=True,numbers={'GT': 3})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Letus take a look what is in there. Wee need to pay attention to the GT number and sample number\n",
    "callset = np.load('example.npz')\n",
    "\n",
    "#how many variants a sample can have. We set this number as the max \n",
    "max_alt = np.max(callset['variants/numalt'])\n",
    "#Transform these data into a dataframe called \"data\"\n",
    "data = allel.vcf_to_dataframe(filename, fields='*', alt_number=6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALT_2</th>\n",
       "      <th>ALT_3</th>\n",
       "      <th>ALT_4</th>\n",
       "      <th>ALT_5</th>\n",
       "      <th>ALT_6</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>NS</th>\n",
       "      <th>DP</th>\n",
       "      <th>DPB</th>\n",
       "      <th>AC_1</th>\n",
       "      <th>AC_2</th>\n",
       "      <th>AC_3</th>\n",
       "      <th>AC_4</th>\n",
       "      <th>AC_5</th>\n",
       "      <th>AC_6</th>\n",
       "      <th>AN</th>\n",
       "      <th>AF_1</th>\n",
       "      <th>AF_2</th>\n",
       "      <th>AF_3</th>\n",
       "      <th>AF_4</th>\n",
       "      <th>AF_5</th>\n",
       "      <th>AF_6</th>\n",
       "      <th>RO</th>\n",
       "      <th>AO_1</th>\n",
       "      <th>AO_2</th>\n",
       "      <th>AO_3</th>\n",
       "      <th>AO_4</th>\n",
       "      <th>AO_5</th>\n",
       "      <th>AO_6</th>\n",
       "      <th>PRO</th>\n",
       "      <th>PAO_1</th>\n",
       "      <th>PAO_2</th>\n",
       "      <th>PAO_3</th>\n",
       "      <th>PAO_4</th>\n",
       "      <th>PAO_5</th>\n",
       "      <th>PAO_6</th>\n",
       "      <th>QR</th>\n",
       "      <th>QA_1</th>\n",
       "      <th>QA_2</th>\n",
       "      <th>QA_3</th>\n",
       "      <th>QA_4</th>\n",
       "      <th>QA_5</th>\n",
       "      <th>QA_6</th>\n",
       "      <th>PQR</th>\n",
       "      <th>PQA_1</th>\n",
       "      <th>PQA_2</th>\n",
       "      <th>PQA_3</th>\n",
       "      <th>PQA_4</th>\n",
       "      <th>PQA_5</th>\n",
       "      <th>PQA_6</th>\n",
       "      <th>SRF</th>\n",
       "      <th>SRR</th>\n",
       "      <th>SAF_1</th>\n",
       "      <th>SAF_2</th>\n",
       "      <th>SAF_3</th>\n",
       "      <th>SAF_4</th>\n",
       "      <th>SAF_5</th>\n",
       "      <th>SAF_6</th>\n",
       "      <th>SAR_1</th>\n",
       "      <th>SAR_2</th>\n",
       "      <th>SAR_3</th>\n",
       "      <th>SAR_4</th>\n",
       "      <th>SAR_5</th>\n",
       "      <th>SAR_6</th>\n",
       "      <th>SRP</th>\n",
       "      <th>SAP_1</th>\n",
       "      <th>SAP_2</th>\n",
       "      <th>SAP_3</th>\n",
       "      <th>SAP_4</th>\n",
       "      <th>SAP_5</th>\n",
       "      <th>SAP_6</th>\n",
       "      <th>AB_1</th>\n",
       "      <th>AB_2</th>\n",
       "      <th>AB_3</th>\n",
       "      <th>AB_4</th>\n",
       "      <th>AB_5</th>\n",
       "      <th>AB_6</th>\n",
       "      <th>ABP_1</th>\n",
       "      <th>ABP_2</th>\n",
       "      <th>ABP_3</th>\n",
       "      <th>ABP_4</th>\n",
       "      <th>ABP_5</th>\n",
       "      <th>ABP_6</th>\n",
       "      <th>RUN_1</th>\n",
       "      <th>RUN_2</th>\n",
       "      <th>RUN_3</th>\n",
       "      <th>RUN_4</th>\n",
       "      <th>RUN_5</th>\n",
       "      <th>RUN_6</th>\n",
       "      <th>RPP_1</th>\n",
       "      <th>RPP_2</th>\n",
       "      <th>RPP_3</th>\n",
       "      <th>RPP_4</th>\n",
       "      <th>RPP_5</th>\n",
       "      <th>RPP_6</th>\n",
       "      <th>RPPR</th>\n",
       "      <th>RPL_1</th>\n",
       "      <th>RPL_2</th>\n",
       "      <th>RPL_3</th>\n",
       "      <th>RPL_4</th>\n",
       "      <th>RPL_5</th>\n",
       "      <th>RPL_6</th>\n",
       "      <th>RPR_1</th>\n",
       "      <th>RPR_2</th>\n",
       "      <th>RPR_3</th>\n",
       "      <th>RPR_4</th>\n",
       "      <th>RPR_5</th>\n",
       "      <th>RPR_6</th>\n",
       "      <th>EPP_1</th>\n",
       "      <th>EPP_2</th>\n",
       "      <th>EPP_3</th>\n",
       "      <th>EPP_4</th>\n",
       "      <th>EPP_5</th>\n",
       "      <th>EPP_6</th>\n",
       "      <th>EPPR</th>\n",
       "      <th>DPRA_1</th>\n",
       "      <th>DPRA_2</th>\n",
       "      <th>DPRA_3</th>\n",
       "      <th>DPRA_4</th>\n",
       "      <th>DPRA_5</th>\n",
       "      <th>DPRA_6</th>\n",
       "      <th>ODDS</th>\n",
       "      <th>GTI</th>\n",
       "      <th>TYPE_1</th>\n",
       "      <th>TYPE_2</th>\n",
       "      <th>TYPE_3</th>\n",
       "      <th>TYPE_4</th>\n",
       "      <th>TYPE_5</th>\n",
       "      <th>TYPE_6</th>\n",
       "      <th>CIGAR_1</th>\n",
       "      <th>CIGAR_2</th>\n",
       "      <th>CIGAR_3</th>\n",
       "      <th>CIGAR_4</th>\n",
       "      <th>CIGAR_5</th>\n",
       "      <th>CIGAR_6</th>\n",
       "      <th>NUMALT</th>\n",
       "      <th>MEANALT_1</th>\n",
       "      <th>MEANALT_2</th>\n",
       "      <th>MEANALT_3</th>\n",
       "      <th>MEANALT_4</th>\n",
       "      <th>MEANALT_5</th>\n",
       "      <th>MEANALT_6</th>\n",
       "      <th>LEN_1</th>\n",
       "      <th>LEN_2</th>\n",
       "      <th>LEN_3</th>\n",
       "      <th>LEN_4</th>\n",
       "      <th>LEN_5</th>\n",
       "      <th>LEN_6</th>\n",
       "      <th>MQM_1</th>\n",
       "      <th>MQM_2</th>\n",
       "      <th>MQM_3</th>\n",
       "      <th>MQM_4</th>\n",
       "      <th>MQM_5</th>\n",
       "      <th>MQM_6</th>\n",
       "      <th>MQMR</th>\n",
       "      <th>PAIRED_1</th>\n",
       "      <th>PAIRED_2</th>\n",
       "      <th>PAIRED_3</th>\n",
       "      <th>PAIRED_4</th>\n",
       "      <th>PAIRED_5</th>\n",
       "      <th>PAIRED_6</th>\n",
       "      <th>PAIREDR</th>\n",
       "      <th>MIN_DP</th>\n",
       "      <th>END</th>\n",
       "      <th>FILTER_PASS</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>altlen_2</th>\n",
       "      <th>altlen_3</th>\n",
       "      <th>altlen_4</th>\n",
       "      <th>altlen_5</th>\n",
       "      <th>altlen_6</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>931393</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.179380e-13</td>\n",
       "      <td>2</td>\n",
       "      <td>4124</td>\n",
       "      <td>4124.0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4029</td>\n",
       "      <td>95</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>160284</td>\n",
       "      <td>3774</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1663</td>\n",
       "      <td>2366</td>\n",
       "      <td>40</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>55</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>269.368988</td>\n",
       "      <td>8.15326</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>4.13032</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>101.278</td>\n",
       "      <td>51.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.61615</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>316.776001</td>\n",
       "      <td>0.999031</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>591.289978</td>\n",
       "      <td>0</td>\n",
       "      <td>snp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1X</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>59.7579</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>65.227402</td>\n",
       "      <td>0.989474</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.966741</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM     POS ID REF ALT_1 ALT_2 ALT_3 ALT_4 ALT_5 ALT_6          QUAL  NS  \\\n",
       "0     1  931393  .   G     T   NaN   NaN   NaN   NaN   NaN  2.179380e-13   2   \n",
       "\n",
       "     DP     DPB  AC_1  AC_2  AC_3  AC_4  AC_5  AC_6  AN  AF_1  AF_2  AF_3  \\\n",
       "0  4124  4124.0     0    -1    -1    -1    -1    -1   6   0.0   NaN   NaN   \n",
       "\n",
       "   AF_4  AF_5  AF_6    RO  AO_1  AO_2  AO_3  AO_4  AO_5  AO_6  PRO  PAO_1  \\\n",
       "0   NaN   NaN   NaN  4029    95    -1    -1    -1    -1    -1  0.0    0.0   \n",
       "\n",
       "   PAO_2  PAO_3  PAO_4  PAO_5  PAO_6      QR  QA_1  QA_2  QA_3  QA_4  QA_5  \\\n",
       "0    NaN    NaN    NaN    NaN    NaN  160284  3774    -1    -1    -1    -1   \n",
       "\n",
       "   QA_6  PQR  PQA_1  PQA_2  PQA_3  PQA_4  PQA_5  PQA_6   SRF   SRR  SAF_1  \\\n",
       "0    -1  0.0    0.0    NaN    NaN    NaN    NaN    NaN  1663  2366     40   \n",
       "\n",
       "   SAF_2  SAF_3  SAF_4  SAF_5  SAF_6  SAR_1  SAR_2  SAR_3  SAR_4  SAR_5  \\\n",
       "0     -1     -1     -1     -1     -1     55     -1     -1     -1     -1   \n",
       "\n",
       "   SAR_6         SRP    SAP_1  SAP_2  SAP_3  SAP_4  SAP_5  SAP_6  AB_1  AB_2  \\\n",
       "0     -1  269.368988  8.15326    NaN    NaN    NaN    NaN    NaN   0.0   NaN   \n",
       "\n",
       "   AB_3  AB_4  AB_5  AB_6  ABP_1  ABP_2  ABP_3  ABP_4  ABP_5  ABP_6  RUN_1  \\\n",
       "0   NaN   NaN   NaN   NaN    0.0    NaN    NaN    NaN    NaN    NaN      1   \n",
       "\n",
       "   RUN_2  RUN_3  RUN_4  RUN_5  RUN_6    RPP_1  RPP_2  RPP_3  RPP_4  RPP_5  \\\n",
       "0     -1     -1     -1     -1     -1  4.13032    NaN    NaN    NaN    NaN   \n",
       "\n",
       "   RPP_6     RPPR  RPL_1  RPL_2  RPL_3  RPL_4  RPL_5  RPL_6  RPR_1  RPR_2  \\\n",
       "0    NaN  101.278   51.0    NaN    NaN    NaN    NaN    NaN   44.0    NaN   \n",
       "\n",
       "   RPR_3  RPR_4  RPR_5  RPR_6    EPP_1  EPP_2  EPP_3  EPP_4  EPP_5  EPP_6  \\\n",
       "0    NaN    NaN    NaN    NaN  9.61615    NaN    NaN    NaN    NaN    NaN   \n",
       "\n",
       "         EPPR    DPRA_1  DPRA_2  DPRA_3  DPRA_4  DPRA_5  DPRA_6        ODDS  \\\n",
       "0  316.776001  0.999031     NaN     NaN     NaN     NaN     NaN  591.289978   \n",
       "\n",
       "   GTI TYPE_1 TYPE_2 TYPE_3 TYPE_4 TYPE_5 TYPE_6 CIGAR_1 CIGAR_2 CIGAR_3  \\\n",
       "0    0    snp    NaN    NaN    NaN    NaN    NaN      1X     NaN     NaN   \n",
       "\n",
       "  CIGAR_4 CIGAR_5 CIGAR_6  NUMALT  MEANALT_1  MEANALT_2  MEANALT_3  MEANALT_4  \\\n",
       "0     NaN     NaN     NaN       1        1.0        NaN        NaN        NaN   \n",
       "\n",
       "   MEANALT_5  MEANALT_6  LEN_1  LEN_2  LEN_3  LEN_4  LEN_5  LEN_6    MQM_1  \\\n",
       "0        NaN        NaN      1     -1     -1     -1     -1     -1  59.7579   \n",
       "\n",
       "   MQM_2  MQM_3  MQM_4  MQM_5  MQM_6       MQMR  PAIRED_1  PAIRED_2  PAIRED_3  \\\n",
       "0    NaN    NaN    NaN    NaN    NaN  65.227402  0.989474       NaN       NaN   \n",
       "\n",
       "   PAIRED_4  PAIRED_5  PAIRED_6   PAIREDR  MIN_DP  END  FILTER_PASS  numalt  \\\n",
       "0       NaN       NaN       NaN  0.966741      -1   -1        False       1   \n",
       "\n",
       "   altlen_1  altlen_2  altlen_3  altlen_4  altlen_5  altlen_6  is_snp  \n",
       "0         0         0         0         0         0         0    True  "
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# A example\n",
    "data[0:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def variant_summary( df=data):\n",
    "'''\n",
    "1. Here. I made a function that take the dataframe as input and extract the following information\n",
    "    'variant_type'         ---To answer Question 1:Type of variation\n",
    "    'variant_allele',      ---What is the variant basepair\n",
    "    'variant_allele_count' ---To answer Question 3:Number of reads supporting the variant\n",
    "    'variant_AF':          ---Variant allele frequency\n",
    "    'variant_vs_ref%'      ---Percentage of reads supporting the variant versus those supporting reference reads\n",
    "'''\n",
    "    #make an empty df\n",
    "    summary = pd.DataFrame(columns=['variant_type',\n",
    "                                    'variant_allele', \n",
    "                                    'variant_allele_count', \n",
    "                                    'variant_AF',\n",
    "                                    'variant_vs_ref%']) \n",
    "       \n",
    "      \n",
    "    \n",
    "    ref_column_index = df.columns.get_loc(\"REF\") # find the reference column from data indexing\n",
    "    ref_read_index = df.columns.get_loc(\"RO\")# find the Reference allele observation count coloumn from data indexing\n",
    "    AF_index =  df.columns.get_loc(\"AF_1\")# find the variant allele observation countcoloumn from data index\n",
    "    \n",
    "    \n",
    "    total_row = df.shape[0] # this is the total case number we need to process\n",
    "    #========================================================================\n",
    "    for row in range(total_row): # run each case\n",
    "        case_call = []\n",
    "        ref_allele = df.iloc[row]['REF'] # the REF allele sequence\n",
    "        \n",
    "        # make a list of variant_type, number of read, and AF for each variant    \n",
    "        for number in range(df.iloc[row]['numalt']): #how many different vatiants this row has\n",
    "            \n",
    "            #Here we go through every different variants\n",
    "            variant_allele = df.iloc[row, ref_column_index+number +1] # tge n-th variant allele\n",
    "            variant_reading = df.iloc[row, ref_read_index +number +1] # match this variant to its allele read count\n",
    "            variant_frequency = df.iloc[row, AF_index+number] # match variant to its AF\n",
    "            variant_vs_ref = round((variant_reading/df.iloc[row]['RO']*100),2) # the % of variant vs ref count\n",
    "            \n",
    "            #Use a list to collect the information of one variant in this row \n",
    "            call = [variant_allele,variant_reading, variant_frequency, variant_vs_ref]\n",
    "            #Use allel length to determine its type            \n",
    "            if len(ref_allele) - len(variant_allele)== 0: #when ref and variant have the same length\n",
    "                call.insert(0, 'Subtitution')\n",
    "            if len(ref_allele) - len(variant_allele) < 0: #when ref is shorter than variant \n",
    "                call.insert(0, 'Insertion')\n",
    "            if len(ref_allele) - len(variant_allele) > 0: # When ref is longer than variant  \n",
    "                call.insert(0, 'Deletion')\n",
    "            # this is a list of list contains all the variant info in this row\n",
    "            case_call.append(call) \n",
    "    #=======================================================================\n",
    "    '''\n",
    "    2. The second step is to analyze the case that has single or multiple variants. \n",
    "        !!! The Task says \"If there are multiple possibilities, annotate with the most deleterious possibility.\" \n",
    "        Quantify the level of varient deleteriousness is a big challenge in cancer genetics, which will take a few more decades.\n",
    "        With my 15+ year of experience in cancer genetics and cancer biology, I simply cannot tell variant A is more deleterious\n",
    "        than variant B without the meeting the patients in clinic and seeing their medical record.\n",
    "\n",
    "        So I will simplify the task and artificially define deleteriousness as Deletion > Insertion > Substitution \n",
    "\n",
    "        To determine whether the variant is Silent, it require a more complex algorithm that consider the coding frame, \n",
    "        overlaping genes, genes on the reverse allele, etc. SO I will not put this catergory here.\n",
    "    '''            \n",
    "       # sorted the variant types by s dummp order\n",
    "        sort_order = {\"Deletion\":0, \"Insertion\":1, \"Subtitution\":2, \"SNP\":3}\n",
    "        # In this list of list, the first item is always the the variant_type. We use it as a sorting role\n",
    "        case_call.sort(key=lambda val:sort_order[val[0]])\n",
    "        final_report = (case_call[0]) # Make a  most \"deleterious\" as the final call this this row\n",
    "        \n",
    "        \n",
    "        summary.loc[len(summary)]= final_report\n",
    "\n",
    "    return summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ywang\\AppData\\Local\\Continuum\\anaconda3\\lib\\site-packages\\ipykernel_launcher.py:31: RuntimeWarning: divide by zero encountered in longlong_scalars\n"
     ]
    }
   ],
   "source": [
    "report = variant_summary() \n",
    "# Just make a copy with better name. This df now has five columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Let's add a few more coulmns here with useful info\n",
    "report.insert(1,'ref_allele', data['REF'])       # The ref allele sequence\n",
    "report.insert(3,'ref_allele_count', data['RO'])  # ref allele count number\n",
    "report['DP'] = data['DP']                        # Answer Question 3: Depth of sequence coverage at the site of variation.\n",
    "report.insert(1,'CHROM', data['CHROM'])          # The location of variant = Chromosome number and position\n",
    "report.insert(2,'POS', data['POS'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Let's make a string for query on the ExAC API\n",
    "report['Query_name'] =   report['CHROM'] +'-' \n",
    "                       + report['POS'].astype(str)+ '-' \n",
    "                       + report['ref_allele'] + '-' \n",
    "                       + report['variant_allele']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "report['AF_ExAC'] = np.nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_type</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>variant_allele</th>\n",
       "      <th>ref_allele_count</th>\n",
       "      <th>variant_allele_count</th>\n",
       "      <th>variant_AF</th>\n",
       "      <th>variant_vs_ref%</th>\n",
       "      <th>DP</th>\n",
       "      <th>Query_name</th>\n",
       "      <th>AF_ExAC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>931393</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>4029</td>\n",
       "      <td>95</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.360000</td>\n",
       "      <td>4124</td>\n",
       "      <td>1-931393-G-T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>935222</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>480</td>\n",
       "      <td>652</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>135.830000</td>\n",
       "      <td>1134</td>\n",
       "      <td>1-935222-C-A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1277533</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>786</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>786</td>\n",
       "      <td>1-1277533-T-C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1284490</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>228</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>228</td>\n",
       "      <td>1-1284490-G-A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1571850</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3961</td>\n",
       "      <td>94</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.370000</td>\n",
       "      <td>4055</td>\n",
       "      <td>1-1571850-G-A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1572579</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3430</td>\n",
       "      <td>26</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.760000</td>\n",
       "      <td>3456</td>\n",
       "      <td>1-1572579-A-G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1575616</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>1118</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>1118</td>\n",
       "      <td>1-1575616-T-C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1575784</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>3206</td>\n",
       "      <td>266</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>8.300000</td>\n",
       "      <td>3476</td>\n",
       "      <td>1-1575784-C-T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1577180</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>40</td>\n",
       "      <td>50</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>125.000000</td>\n",
       "      <td>90</td>\n",
       "      <td>1-1577180-C-T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1635004</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>4076</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>4076</td>\n",
       "      <td>1-1635004-T-C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  variant_type CHROM      POS ref_allele variant_allele  ref_allele_count  \\\n",
       "0  Subtitution     1   931393          G              T              4029   \n",
       "1  Subtitution     1   935222          C              A               480   \n",
       "2  Subtitution     1  1277533          T              C                 0   \n",
       "3  Subtitution     1  1284490          G              A                 0   \n",
       "4  Subtitution     1  1571850          G              A              3961   \n",
       "5  Subtitution     1  1572579          A              G              3430   \n",
       "6  Subtitution     1  1575616          T              C                 0   \n",
       "7  Subtitution     1  1575784          C              T              3206   \n",
       "8  Subtitution     1  1577180          C              T                40   \n",
       "9  Subtitution     1  1635004          T              C                 0   \n",
       "\n",
       "  variant_allele_count  variant_AF  variant_vs_ref%    DP     Query_name  \\\n",
       "0                   95    0.000000         2.360000  4124   1-931393-G-T   \n",
       "1                  652    0.666667       135.830000  1134   1-935222-C-A   \n",
       "2                  786    1.000000              inf   786  1-1277533-T-C   \n",
       "3                  228    1.000000              inf   228  1-1284490-G-A   \n",
       "4                   94    0.000000         2.370000  4055  1-1571850-G-A   \n",
       "5                   26    0.000000         0.760000  3456  1-1572579-A-G   \n",
       "6                 1118    1.000000              inf  1118  1-1575616-T-C   \n",
       "7                  266    0.333333         8.300000  3476  1-1575784-C-T   \n",
       "8                   50    0.666667       125.000000    90  1-1577180-C-T   \n",
       "9                 4076    1.000000              inf  4076  1-1635004-T-C   \n",
       "\n",
       "   AF_ExAC  \n",
       "0      NaN  \n",
       "1      NaN  \n",
       "2      NaN  \n",
       "3      NaN  \n",
       "4      NaN  \n",
       "5      NaN  \n",
       "6      NaN  \n",
       "7      NaN  \n",
       "8      NaN  \n",
       "9      NaN  "
      ]
     },
     "execution_count": 205,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Take a look the df by far\n",
    "report[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'allele_count': 42,\n",
      " 'allele_freq': 0.4883720930232558,\n",
      " 'allele_num': 86,\n",
      " 'alt': 'T',\n",
      " 'chrom': '1',\n",
      " 'filter': 'VQSRTrancheSNP99.80to99.90',\n",
      " 'genes': ['ENSG00000248333'],\n",
      " 'genotype_depths': [[[2.5, 1801],\n",
      "                      [7.5, 271],\n",
      "                      [12.5, 40],\n",
      "                      [17.5, 5],\n",
      "                      [22.5, 1],\n",
      "                      [27.5, 0],\n",
      "                      [32.5, 1],\n",
      "                      [37.5, 0],\n",
      "                      [42.5, 0],\n",
      "                      [47.5, 0],\n",
      "                      [52.5, 0],\n",
      "                      [57.5, 0],\n",
      "                      [62.5, 0],\n",
      "                      [67.5, 0],\n",
      "                      [72.5, 0],\n",
      "                      [77.5, 0],\n",
      "                      [82.5, 0],\n",
      "                      [87.5, 0],\n",
      "                      [92.5, 0],\n",
      "                      [97.5, 0]],\n",
      "                     [[2.5, 115],\n",
      "                      [7.5, 107],\n",
      "                      [12.5, 35],\n",
      "                      [17.5, 5],\n",
      "                      [22.5, 1],\n",
      "                      [27.5, 0],\n",
      "                      [32.5, 1],\n",
      "                      [37.5, 0],\n",
      "                      [42.5, 0],\n",
      "                      [47.5, 0],\n",
      "                      [52.5, 0],\n",
      "                      [57.5, 0],\n",
      "                      [62.5, 0],\n",
      "                      [67.5, 0],\n",
      "                      [72.5, 0],\n",
      "                      [77.5, 0],\n",
      "                      [82.5, 0],\n",
      "                      [87.5, 0],\n",
      "                      [92.5, 0],\n",
      "                      [97.5, 0]]],\n",
      " 'genotype_qualities': [[[2.5, 641],\n",
      "                         [7.5, 1218],\n",
      "                         [12.5, 44],\n",
      "                         [17.5, 53],\n",
      "                         [22.5, 29],\n",
      "                         [27.5, 4],\n",
      "                         [32.5, 6],\n",
      "                         [37.5, 29],\n",
      "                         [42.5, 9],\n",
      "                         [47.5, 7],\n",
      "                         [52.5, 8],\n",
      "                         [57.5, 28],\n",
      "                         [62.5, 10],\n",
      "                         [67.5, 4],\n",
      "                         [72.5, 6],\n",
      "                         [77.5, 4],\n",
      "                         [82.5, 5],\n",
      "                         [87.5, 1],\n",
      "                         [92.5, 1],\n",
      "                         [97.5, 12]],\n",
      "                        [[2.5, 0],\n",
      "                         [7.5, 49],\n",
      "                         [12.5, 31],\n",
      "                         [17.5, 50],\n",
      "                         [22.5, 2],\n",
      "                         [27.5, 3],\n",
      "                         [32.5, 5],\n",
      "                         [37.5, 29],\n",
      "                         [42.5, 9],\n",
      "                         [47.5, 7],\n",
      "                         [52.5, 8],\n",
      "                         [57.5, 28],\n",
      "                         [62.5, 10],\n",
      "                         [67.5, 4],\n",
      "                         [72.5, 6],\n",
      "                         [77.5, 4],\n",
      "                         [82.5, 5],\n",
      "                         [87.5, 1],\n",
      "                         [92.5, 1],\n",
      "                         [97.5, 12]]],\n",
      " 'hom_count': 0,\n",
      " 'orig_alt_alleles': ['1-1577180-C-T'],\n",
      " 'pop_acs': {'African': 0,\n",
      "             'East Asian': 4,\n",
      "             'European (Finnish)': 0,\n",
      "             'European (Non-Finnish)': 35,\n",
      "             'Latino': 0,\n",
      "             'Other': 1,\n",
      "             'South Asian': 2},\n",
      " 'pop_ans': {'African': 0,\n",
      "             'East Asian': 8,\n",
      "             'European (Finnish)': 0,\n",
      "             'European (Non-Finnish)': 72,\n",
      "             'Latino': 0,\n",
      "             'Other': 2,\n",
      "             'South Asian': 4},\n",
      " 'pop_homs': {'African': 0,\n",
      "              'East Asian': 0,\n",
      "              'European (Finnish)': 0,\n",
      "              'European (Non-Finnish)': 0,\n",
      "              'Latino': 0,\n",
      "              'Other': 0,\n",
      "              'South Asian': 0},\n",
      " 'pos': 1577180,\n",
      " 'quality_metrics': {'BaseQRankSum': '0.937',\n",
      "                     'ClippingRankSum': '0.361',\n",
      "                     'DP': '8082',\n",
      "                     'FS': '0.000',\n",
      "                     'InbreedingCoeff': '-0.0362',\n",
      "                     'MQ': '22.27',\n",
      "                     'MQRankSum': '0.546',\n",
      "                     'QD': '12.13',\n",
      "                     'ReadPosRankSum': '0.540',\n",
      "                     'VQSLOD': '-5.559e+00'},\n",
      " 'ref': 'C',\n",
      " 'rsid': 'rs150540375',\n",
      " 'site_quality': 36077.84,\n",
      " 'transcripts': ['ENST00000513088',\n",
      "                 'ENST00000317673',\n",
      "                 'ENST00000341832',\n",
      "                 'ENST00000407249',\n",
      "                 'ENST00000340677'],\n",
      " 'variant_id': '1-1577180-C-T',\n",
      " 'vep_annotations': [{'AA_MAF': '',\n",
      "                      'AFR_MAF': '',\n",
      "                      'ALLELE_NUM': '1',\n",
      "                      'AMR_MAF': '',\n",
      "                      'ASN_MAF': '',\n",
      "                      'Allele': 'T',\n",
      "                      'Amino_acids': 'A',\n",
      "                      'BIOTYPE': 'protein_coding',\n",
      "                      'CANONICAL': '',\n",
      "                      'CCDS': '',\n",
      "                      'CDS_position': '855',\n",
      "                      'CLIN_SIG': '',\n",
      "                      'Codons': 'gcG/gcA',\n",
      "                      'Consequence': 'splice_region_variant&synonymous_variant',\n",
      "                      'DISTANCE': '',\n",
      "                      'DOMAINS': 'Low_complexity_(Seg):Seg&PROSITE_profiles:PS50313',\n",
      "                      'EA_MAF': '',\n",
      "                      'ENSP': 'ENSP00000463200',\n",
      "                      'EUR_MAF': '',\n",
      "                      'EXON': '10/21',\n",
      "                      'Existing_variation': 'rs150540375',\n",
      "                      'Feature': 'ENST00000317673',\n",
      "                      'Feature_type': 'Transcript',\n",
      "                      'GMAF': '',\n",
      "                      'Gene': 'ENSG00000248333',\n",
      "                      'HGNC_ID': '1729',\n",
      "                      'HGVSc': 'ENST00000317673.7:c.855G>A',\n",
      "                      'HGVSp': 'ENST00000317673.7:c.855G>A(p.%3D)',\n",
      "                      'HIGH_INF_POS': '',\n",
      "                      'INTRON': '',\n",
      "                      'LoF': '',\n",
      "                      'LoF_filter': '',\n",
      "                      'LoF_flags': '',\n",
      "                      'LoF_info': '',\n",
      "                      'MOTIF_NAME': '',\n",
      "                      'MOTIF_POS': '',\n",
      "                      'MOTIF_SCORE_CHANGE': '',\n",
      "                      'PUBMED': '',\n",
      "                      'PolyPhen': '',\n",
      "                      'Protein_position': '285',\n",
      "                      'SIFT': '',\n",
      "                      'SOMATIC': '',\n",
      "                      'STRAND': '-1',\n",
      "                      'SWISSPROT': '',\n",
      "                      'SYMBOL': 'CDK11B',\n",
      "                      'SYMBOL_SOURCE': 'HGNC',\n",
      "                      'TREMBL': 'Q6P5Y5_HUMAN&Q5QPQ9_HUMAN&J3QKR5_HUMAN&A4VCI5_HUMAN',\n",
      "                      'UNIPARC': 'UPI0002064DC2',\n",
      "                      'cDNA_position': '855',\n",
      "                      'major_consequence': 'splice_region_variant'},\n",
      "                     {'AA_MAF': '',\n",
      "                      'AFR_MAF': '',\n",
      "                      'ALLELE_NUM': '1',\n",
      "                      'AMR_MAF': '',\n",
      "                      'ASN_MAF': '',\n",
      "                      'Allele': 'T',\n",
      "                      'Amino_acids': 'A',\n",
      "                      'BIOTYPE': 'protein_coding',\n",
      "                      'CANONICAL': 'YES',\n",
      "                      'CCDS': '',\n",
      "                      'CDS_position': '861',\n",
      "                      'CLIN_SIG': '',\n",
      "                      'Codons': 'gcG/gcA',\n",
      "                      'Consequence': 'splice_region_variant&synonymous_variant',\n",
      "                      'DISTANCE': '',\n",
      "                      'DOMAINS': 'Low_complexity_(Seg):Seg&PROSITE_profiles:PS50313',\n",
      "                      'EA_MAF': '',\n",
      "                      'ENSP': 'ENSP00000464036',\n",
      "                      'EUR_MAF': '',\n",
      "                      'EXON': '10/21',\n",
      "                      'Existing_variation': 'rs150540375',\n",
      "                      'Feature': 'ENST00000407249',\n",
      "                      'Feature_type': 'Transcript',\n",
      "                      'GMAF': '',\n",
      "                      'Gene': 'ENSG00000248333',\n",
      "                      'HGNC_ID': '1729',\n",
      "                      'HGVSc': 'ENST00000407249.3:c.861G>A',\n",
      "                      'HGVSp': 'ENST00000407249.3:c.861G>A(p.%3D)',\n",
      "                      'HIGH_INF_POS': '',\n",
      "                      'INTRON': '',\n",
      "                      'LoF': '',\n",
      "                      'LoF_filter': '',\n",
      "                      'LoF_flags': '',\n",
      "                      'LoF_info': '',\n",
      "                      'MOTIF_NAME': '',\n",
      "                      'MOTIF_POS': '',\n",
      "                      'MOTIF_SCORE_CHANGE': '',\n",
      "                      'PUBMED': '',\n",
      "                      'PolyPhen': '',\n",
      "                      'Protein_position': '287',\n",
      "                      'SIFT': '',\n",
      "                      'SOMATIC': '',\n",
      "                      'STRAND': '-1',\n",
      "                      'SWISSPROT': '',\n",
      "                      'SYMBOL': 'CDK11B',\n",
      "                      'SYMBOL_SOURCE': 'HGNC',\n",
      "                      'TREMBL': 'Q6P5Y5_HUMAN&Q5QPQ9_HUMAN&J3QR44_HUMAN&A4VCI5_HUMAN',\n",
      "                      'UNIPARC': 'UPI0000D61E1A',\n",
      "                      'cDNA_position': '861',\n",
      "                      'major_consequence': 'splice_region_variant'},\n",
      "                     {'AA_MAF': '',\n",
      "                      'AFR_MAF': '',\n",
      "                      'ALLELE_NUM': '1',\n",
      "                      'AMR_MAF': '',\n",
      "                      'ASN_MAF': '',\n",
      "                      'Allele': 'T',\n",
      "                      'Amino_acids': 'A',\n",
      "                      'BIOTYPE': 'protein_coding',\n",
      "                      'CANONICAL': '',\n",
      "                      'CCDS': '',\n",
      "                      'CDS_position': '822',\n",
      "                      'CLIN_SIG': '',\n",
      "                      'Codons': 'gcG/gcA',\n",
      "                      'Consequence': 'splice_region_variant&synonymous_variant',\n",
      "                      'DISTANCE': '',\n",
      "                      'DOMAINS': 'Low_complexity_(Seg):Seg&PROSITE_profiles:PS50313',\n",
      "                      'EA_MAF': '',\n",
      "                      'ENSP': 'ENSP00000464016',\n",
      "                      'EUR_MAF': '',\n",
      "                      'EXON': '9/20',\n",
      "                      'Existing_variation': 'rs150540375',\n",
      "                      'Feature': 'ENST00000340677',\n",
      "                      'Feature_type': 'Transcript',\n",
      "                      'GMAF': '',\n",
      "                      'Gene': 'ENSG00000248333',\n",
      "                      'HGNC_ID': '1729',\n",
      "                      'HGVSc': 'ENST00000340677.5:c.822G>A',\n",
      "                      'HGVSp': 'ENST00000340677.5:c.822G>A(p.%3D)',\n",
      "                      'HIGH_INF_POS': '',\n",
      "                      'INTRON': '',\n",
      "                      'LoF': '',\n",
      "                      'LoF_filter': '',\n",
      "                      'LoF_flags': '',\n",
      "                      'LoF_info': '',\n",
      "                      'MOTIF_NAME': '',\n",
      "                      'MOTIF_POS': '',\n",
      "                      'MOTIF_SCORE_CHANGE': '',\n",
      "                      'PUBMED': '',\n",
      "                      'PolyPhen': '',\n",
      "                      'Protein_position': '274',\n",
      "                      'SIFT': '',\n",
      "                      'SOMATIC': '',\n",
      "                      'STRAND': '-1',\n",
      "                      'SWISSPROT': '',\n",
      "                      'SYMBOL': 'CDK11B',\n",
      "                      'SYMBOL_SOURCE': 'HGNC',\n",
      "                      'TREMBL': 'Q6P5Y5_HUMAN&Q5QPQ9_HUMAN&J3QR29_HUMAN&A4VCI5_HUMAN',\n",
      "                      'UNIPARC': 'UPI0000456D60',\n",
      "                      'cDNA_position': '822',\n",
      "                      'major_consequence': 'splice_region_variant'},\n",
      "                     {'AA_MAF': '',\n",
      "                      'AFR_MAF': '',\n",
      "                      'ALLELE_NUM': '1',\n",
      "                      'AMR_MAF': '',\n",
      "                      'ASN_MAF': '',\n",
      "                      'Allele': 'T',\n",
      "                      'Amino_acids': 'A',\n",
      "                      'BIOTYPE': 'protein_coding',\n",
      "                      'CANONICAL': '',\n",
      "                      'CCDS': '',\n",
      "                      'CDS_position': '720',\n",
      "                      'CLIN_SIG': '',\n",
      "                      'Codons': 'gcG/gcA',\n",
      "                      'Consequence': 'splice_region_variant&synonymous_variant',\n",
      "                      'DISTANCE': '',\n",
      "                      'DOMAINS': 'Low_complexity_(Seg):Seg&PROSITE_profiles:PS50313',\n",
      "                      'EA_MAF': '',\n",
      "                      'ENSP': 'ENSP00000463048',\n",
      "                      'EUR_MAF': '',\n",
      "                      'EXON': '9/20',\n",
      "                      'Existing_variation': 'rs150540375',\n",
      "                      'Feature': 'ENST00000341832',\n",
      "                      'Feature_type': 'Transcript',\n",
      "                      'GMAF': '',\n",
      "                      'Gene': 'ENSG00000248333',\n",
      "                      'HGNC_ID': '1729',\n",
      "                      'HGVSc': 'ENST00000341832.6:c.720G>A',\n",
      "                      'HGVSp': 'ENST00000341832.6:c.720G>A(p.%3D)',\n",
      "                      'HIGH_INF_POS': '',\n",
      "                      'INTRON': '',\n",
      "                      'LoF': '',\n",
      "                      'LoF_filter': '',\n",
      "                      'LoF_flags': '',\n",
      "                      'LoF_info': '',\n",
      "                      'MOTIF_NAME': '',\n",
      "                      'MOTIF_POS': '',\n",
      "                      'MOTIF_SCORE_CHANGE': '',\n",
      "                      'PUBMED': '',\n",
      "                      'PolyPhen': '',\n",
      "                      'Protein_position': '240',\n",
      "                      'SIFT': '',\n",
      "                      'SOMATIC': '',\n",
      "                      'STRAND': '-1',\n",
      "                      'SWISSPROT': '',\n",
      "                      'SYMBOL': 'CDK11B',\n",
      "                      'SYMBOL_SOURCE': 'HGNC',\n",
      "                      'TREMBL': 'Q6P5Y5_HUMAN&J3KTL7_HUMAN&A4VCI5_HUMAN',\n",
      "                      'UNIPARC': 'UPI0000456D61',\n",
      "                      'cDNA_position': '720',\n",
      "                      'major_consequence': 'splice_region_variant'},\n",
      "                     {'AA_MAF': '',\n",
      "                      'AFR_MAF': '',\n",
      "                      'ALLELE_NUM': '1',\n",
      "                      'AMR_MAF': '',\n",
      "                      'ASN_MAF': '',\n",
      "                      'Allele': 'T',\n",
      "                      'Amino_acids': 'A',\n",
      "                      'BIOTYPE': 'protein_coding',\n",
      "                      'CANONICAL': '',\n",
      "                      'CCDS': '',\n",
      "                      'CDS_position': '360',\n",
      "                      'CLIN_SIG': '',\n",
      "                      'Codons': 'gcG/gcA',\n",
      "                      'Consequence': 'splice_region_variant&synonymous_variant',\n",
      "                      'DISTANCE': '',\n",
      "                      'DOMAINS': 'Low_complexity_(Seg):Seg&PROSITE_profiles:PS50313',\n",
      "                      'EA_MAF': '',\n",
      "                      'ENSP': 'ENSP00000463404',\n",
      "                      'EUR_MAF': '',\n",
      "                      'EXON': '4/15',\n",
      "                      'Existing_variation': 'rs150540375',\n",
      "                      'Feature': 'ENST00000513088',\n",
      "                      'Feature_type': 'Transcript',\n",
      "                      'GMAF': '',\n",
      "                      'Gene': 'ENSG00000248333',\n",
      "                      'HGNC_ID': '1729',\n",
      "                      'HGVSc': 'ENST00000513088.2:c.358G>A',\n",
      "                      'HGVSp': 'ENST00000513088.2:c.358G>A(p.%3D)',\n",
      "                      'HIGH_INF_POS': '',\n",
      "                      'INTRON': '',\n",
      "                      'LoF': '',\n",
      "                      'LoF_filter': '',\n",
      "                      'LoF_flags': '',\n",
      "                      'LoF_info': '',\n",
      "                      'MOTIF_NAME': '',\n",
      "                      'MOTIF_POS': '',\n",
      "                      'MOTIF_SCORE_CHANGE': '',\n",
      "                      'PUBMED': '',\n",
      "                      'PolyPhen': '',\n",
      "                      'Protein_position': '120',\n",
      "                      'SIFT': '',\n",
      "                      'SOMATIC': '',\n",
      "                      'STRAND': '-1',\n",
      "                      'SWISSPROT': '',\n",
      "                      'SYMBOL': 'CDK11B',\n",
      "                      'SYMBOL_SOURCE': 'HGNC',\n",
      "                      'TREMBL': 'Q6P5Y5_HUMAN&A4VCI5_HUMAN',\n",
      "                      'UNIPARC': 'UPI000268AE08',\n",
      "                      'cDNA_position': '358',\n",
      "                      'major_consequence': 'splice_region_variant'}],\n",
      " 'xpos': 1001577180,\n",
      " 'xstart': 1001577180,\n",
      " 'xstop': 1001577180}\n"
     ]
    }
   ],
   "source": [
    "#Let's test the ExAC API\n",
    "query_name = 'http://exac.hms.harvard.edu/rest/variant/variant/'+report.loc[8]['Query_name']\n",
    "r = requests.get(query_name)\n",
    "#Use pprint to show a redable version of the query result\n",
    "pprint.pprint(r.json())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.4883720930232558"
      ]
     },
     "execution_count": 258,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# test one example for it AF info\n",
    "test_AF = r.json().get(\"allele_freq\")\n",
    "test_AF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use the query string in loop to query the Allele frequency of variant\n",
    "# Someimtes the ExAC API has timeout, so we use a list incase the query loop is interrupted\n",
    "AF_ExAC =[] \n",
    "for row in range(len(data)):\n",
    "    query_name = 'http://exac.hms.harvard.edu/rest/variant/variant/'+report.loc[row]['Query_name']\n",
    "    r = requests.get(query_name)\n",
    "    ExAC_AF.append((r.json().get(\"allele_freq\")))\n",
    "# If we want, we can query more info such as Gene Symblo, etc.\n",
    "\n",
    "report['AF_ExAC']=AF_ExAC # add this new information as a new column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_type</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>variant_allele</th>\n",
       "      <th>ref_allele_count</th>\n",
       "      <th>variant_allele_count</th>\n",
       "      <th>variant_AF</th>\n",
       "      <th>variant_vs_ref%</th>\n",
       "      <th>DP</th>\n",
       "      <th>Query_name</th>\n",
       "      <th>AF_ExAC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>931393</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>4029</td>\n",
       "      <td>95</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.360000e+00</td>\n",
       "      <td>4124</td>\n",
       "      <td>1-931393-G-T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>935222</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>480</td>\n",
       "      <td>652</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>1.358300e+02</td>\n",
       "      <td>1134</td>\n",
       "      <td>1-935222-C-A</td>\n",
       "      <td>0.661111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1277533</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>786</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>786</td>\n",
       "      <td>1-1277533-T-C</td>\n",
       "      <td>0.997571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1284490</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>228</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>228</td>\n",
       "      <td>1-1284490-G-A</td>\n",
       "      <td>0.897299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1571850</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3961</td>\n",
       "      <td>94</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.370000e+00</td>\n",
       "      <td>4055</td>\n",
       "      <td>1-1571850-G-A</td>\n",
       "      <td>0.000011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1572579</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3430</td>\n",
       "      <td>26</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.600000e-01</td>\n",
       "      <td>3456</td>\n",
       "      <td>1-1572579-A-G</td>\n",
       "      <td>0.002177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1575616</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>1118</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>1118</td>\n",
       "      <td>1-1575616-T-C</td>\n",
       "      <td>0.843984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1575784</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>3206</td>\n",
       "      <td>266</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>8.300000e+00</td>\n",
       "      <td>3476</td>\n",
       "      <td>1-1575784-C-T</td>\n",
       "      <td>0.094886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1577180</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>40</td>\n",
       "      <td>50</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>1.250000e+02</td>\n",
       "      <td>90</td>\n",
       "      <td>1-1577180-C-T</td>\n",
       "      <td>0.488372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1635004</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>4076</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>4076</td>\n",
       "      <td>1-1635004-T-C</td>\n",
       "      <td>0.580487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1635749</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>3060</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.530000e+05</td>\n",
       "      <td>3062</td>\n",
       "      <td>1-1635749-C-A</td>\n",
       "      <td>0.471720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1636044</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>3702</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>3702</td>\n",
       "      <td>1-1636044-G-A</td>\n",
       "      <td>0.564420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1638994</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>696</td>\n",
       "      <td>454</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>6.523000e+01</td>\n",
       "      <td>1150</td>\n",
       "      <td>1-1638994-C-T</td>\n",
       "      <td>0.164064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1647778</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>4492</td>\n",
       "      <td>626</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>1.394000e+01</td>\n",
       "      <td>5118</td>\n",
       "      <td>1-1647778-C-G</td>\n",
       "      <td>0.500670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1647814</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4958</td>\n",
       "      <td>632</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>1.275000e+01</td>\n",
       "      <td>5590</td>\n",
       "      <td>1-1647814-T-C</td>\n",
       "      <td>0.501101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1647871</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4506</td>\n",
       "      <td>640</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>1.420000e+01</td>\n",
       "      <td>5146</td>\n",
       "      <td>1-1647871-T-C</td>\n",
       "      <td>0.500961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1650787</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2512</td>\n",
       "      <td>124</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.940000e+00</td>\n",
       "      <td>2636</td>\n",
       "      <td>1-1650787-T-C</td>\n",
       "      <td>0.501135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1650797</td>\n",
       "      <td>ATTTT</td>\n",
       "      <td>GTTTC</td>\n",
       "      <td>2560</td>\n",
       "      <td>74</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.890000e+00</td>\n",
       "      <td>2636</td>\n",
       "      <td>1-1650797-ATTTT-GTTTC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1650807</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>278</td>\n",
       "      <td>2532</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>9.107900e+02</td>\n",
       "      <td>2810</td>\n",
       "      <td>1-1650807-T-C</td>\n",
       "      <td>0.478368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1650832</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2874</td>\n",
       "      <td>378</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>1.315000e+01</td>\n",
       "      <td>3252</td>\n",
       "      <td>1-1650832-A-G</td>\n",
       "      <td>0.501110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1650845</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>376</td>\n",
       "      <td>2618</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>6.962800e+02</td>\n",
       "      <td>2994</td>\n",
       "      <td>1-1650845-G-A</td>\n",
       "      <td>0.478357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1653004</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0</td>\n",
       "      <td>2376</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>2376</td>\n",
       "      <td>1-1653004-T-C</td>\n",
       "      <td>0.848546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>1653028</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>2722</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>2722</td>\n",
       "      <td>1-1653028-C-T</td>\n",
       "      <td>0.854939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>1</td>\n",
       "      <td>1654129</td>\n",
       "      <td>TAAAAAAAT</td>\n",
       "      <td>TAAAAAAT</td>\n",
       "      <td>2116</td>\n",
       "      <td>80</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.780000e+00</td>\n",
       "      <td>2198</td>\n",
       "      <td>1-1654129-TAAAAAAAT-TAAAAAAT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>2160735</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>2061</td>\n",
       "      <td>50</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.430000e+00</td>\n",
       "      <td>2111</td>\n",
       "      <td>1-2160735-T-A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>2164935</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1950</td>\n",
       "      <td>1818</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>9.323000e+01</td>\n",
       "      <td>3772</td>\n",
       "      <td>1-2164935-G-T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>2488153</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1002</td>\n",
       "      <td>998</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>9.960000e+01</td>\n",
       "      <td>2000</td>\n",
       "      <td>1-2488153-A-G</td>\n",
       "      <td>0.544314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>2489247</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1593</td>\n",
       "      <td>41</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.570000e+00</td>\n",
       "      <td>1634</td>\n",
       "      <td>1-2489247-C-G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>2985885</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>108</td>\n",
       "      <td>74</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>6.852000e+01</td>\n",
       "      <td>184</td>\n",
       "      <td>1-2985885-C-G</td>\n",
       "      <td>0.460045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>1</td>\n",
       "      <td>3301721</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>646</td>\n",
       "      <td>686</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>1.061900e+02</td>\n",
       "      <td>1332</td>\n",
       "      <td>1-3301721-C-T</td>\n",
       "      <td>0.329498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6947</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>X</td>\n",
       "      <td>135960146</td>\n",
       "      <td>GA</td>\n",
       "      <td>GAAA</td>\n",
       "      <td>3836</td>\n",
       "      <td>390</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>1.017000e+01</td>\n",
       "      <td>4228</td>\n",
       "      <td>X-135960146-GA-GAAA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6948</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>135960230</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2624</td>\n",
       "      <td>118</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.500000e+00</td>\n",
       "      <td>2742</td>\n",
       "      <td>X-135960230-C-T</td>\n",
       "      <td>0.134586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6949</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>X</td>\n",
       "      <td>135960247</td>\n",
       "      <td>TAAAAAAAAAG</td>\n",
       "      <td>TAAAAAAAAG</td>\n",
       "      <td>1780</td>\n",
       "      <td>34</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.910000e+00</td>\n",
       "      <td>1860</td>\n",
       "      <td>X-135960247-TAAAAAAAAAG-TAAAAAAAAG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6950</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>X</td>\n",
       "      <td>135961585</td>\n",
       "      <td>ATGTT</td>\n",
       "      <td>ATTT</td>\n",
       "      <td>0</td>\n",
       "      <td>252</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>inf</td>\n",
       "      <td>1306</td>\n",
       "      <td>X-135961585-ATGTT-ATTT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6951</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>X</td>\n",
       "      <td>147582595</td>\n",
       "      <td>CCCGCCGCCGCCGC</td>\n",
       "      <td>CCCGCCGCCGC</td>\n",
       "      <td>800</td>\n",
       "      <td>26</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.250000e+00</td>\n",
       "      <td>848</td>\n",
       "      <td>X-147582595-CCCGCCGCCGCCGC-CCCGCCGCCGC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6952</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>147919280</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1050</td>\n",
       "      <td>1046</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>9.962000e+01</td>\n",
       "      <td>2096</td>\n",
       "      <td>X-147919280-T-G</td>\n",
       "      <td>0.006129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6953</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>148035200</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>828</td>\n",
       "      <td>586</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>7.077000e+01</td>\n",
       "      <td>1416</td>\n",
       "      <td>X-148035200-G-A</td>\n",
       "      <td>0.073504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6954</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>X</td>\n",
       "      <td>148048333</td>\n",
       "      <td>CAAAAAAG</td>\n",
       "      <td>CAAAAAG</td>\n",
       "      <td>4740</td>\n",
       "      <td>24</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.100000e-01</td>\n",
       "      <td>4784</td>\n",
       "      <td>X-148048333-CAAAAAAG-CAAAAAG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6955</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>148055144</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0</td>\n",
       "      <td>2194</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>2194</td>\n",
       "      <td>X-148055144-A-G</td>\n",
       "      <td>0.982914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6956</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>X</td>\n",
       "      <td>149639632</td>\n",
       "      <td>TGCAGCAGCAGCAGCAGCAA</td>\n",
       "      <td>TGCAGCAGCAGCAGCAA</td>\n",
       "      <td>2654</td>\n",
       "      <td>72</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.710000e+00</td>\n",
       "      <td>2772</td>\n",
       "      <td>X-149639632-TGCAGCAGCAGCAGCAGCAA-TGCAGCAGCAGCA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6957</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>X</td>\n",
       "      <td>149639651</td>\n",
       "      <td>ACAGCAGCAGCAGC</td>\n",
       "      <td>ACAGCAGCAGC</td>\n",
       "      <td>2854</td>\n",
       "      <td>18</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.300000e-01</td>\n",
       "      <td>2894</td>\n",
       "      <td>X-149639651-ACAGCAGCAGCAGC-ACAGCAGCAGC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6958</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>149653540</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1190</td>\n",
       "      <td>862</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>7.244000e+01</td>\n",
       "      <td>2052</td>\n",
       "      <td>X-149653540-G-A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6959</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>149653616</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>2198</td>\n",
       "      <td>1630</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>7.416000e+01</td>\n",
       "      <td>3832</td>\n",
       "      <td>X-149653616-G-T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6960</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>153278829</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>650</td>\n",
       "      <td>590</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>9.077000e+01</td>\n",
       "      <td>1240</td>\n",
       "      <td>X-153278829-G-A</td>\n",
       "      <td>0.685193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6961</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>153283470</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2854</td>\n",
       "      <td>20</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.000000e-01</td>\n",
       "      <td>2874</td>\n",
       "      <td>X-153283470-C-T</td>\n",
       "      <td>0.000012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6962</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>153284192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1110</td>\n",
       "      <td>1068</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>9.622000e+01</td>\n",
       "      <td>2178</td>\n",
       "      <td>X-153284192-A-G</td>\n",
       "      <td>0.750428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6963</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>153284483</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>606</td>\n",
       "      <td>652</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>1.075900e+02</td>\n",
       "      <td>1258</td>\n",
       "      <td>X-153284483-G-A</td>\n",
       "      <td>0.652726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6964</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>153994596</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>3508</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.754000e+05</td>\n",
       "      <td>3512</td>\n",
       "      <td>X-153994596-G-T</td>\n",
       "      <td>0.997436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6965</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>X</td>\n",
       "      <td>154005088</td>\n",
       "      <td>CAAGAAGAAGAAGAAGAAGAAGA</td>\n",
       "      <td>CAAGAAGAAGAAGAAGAAGA</td>\n",
       "      <td>1124</td>\n",
       "      <td>76</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.760000e+00</td>\n",
       "      <td>1206</td>\n",
       "      <td>X-154005088-CAAGAAGAAGAAGAAGAAGAAGA-CAAGAAGAAG...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6966</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>154294900</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1778</td>\n",
       "      <td>1512</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>8.504000e+01</td>\n",
       "      <td>3296</td>\n",
       "      <td>X-154294900-C-A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6967</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>154998442</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1438</td>\n",
       "      <td>1210</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>8.414000e+01</td>\n",
       "      <td>2648</td>\n",
       "      <td>X-154998442-G-C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6968</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>155003886</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3030</td>\n",
       "      <td>830</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>2.739000e+01</td>\n",
       "      <td>3860</td>\n",
       "      <td>X-155003886-G-A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6969</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>155004280</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2252</td>\n",
       "      <td>2212</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>9.822000e+01</td>\n",
       "      <td>4464</td>\n",
       "      <td>X-155004280-A-G</td>\n",
       "      <td>0.214952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6970</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>155007278</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0</td>\n",
       "      <td>520</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>inf</td>\n",
       "      <td>520</td>\n",
       "      <td>X-155007278-G-A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6971</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>155009661</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>4028</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.014000e+05</td>\n",
       "      <td>4030</td>\n",
       "      <td>X-155009661-A-C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6972</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>155011829</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>6</td>\n",
       "      <td>1556</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.593333e+04</td>\n",
       "      <td>1562</td>\n",
       "      <td>X-155011829-T-G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6973</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>155011926</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4</td>\n",
       "      <td>2798</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>6.995000e+04</td>\n",
       "      <td>2802</td>\n",
       "      <td>X-155011926-T-C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6974</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>155233098</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>522</td>\n",
       "      <td>460</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>8.812000e+01</td>\n",
       "      <td>982</td>\n",
       "      <td>X-155233098-T-C</td>\n",
       "      <td>0.448144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6975</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>155239499</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>442</td>\n",
       "      <td>46</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>1.041000e+01</td>\n",
       "      <td>488</td>\n",
       "      <td>X-155239499-G-A</td>\n",
       "      <td>0.005948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6976</th>\n",
       "      <td>Subtitution</td>\n",
       "      <td>X</td>\n",
       "      <td>155239824</td>\n",
       "      <td>ACAA</td>\n",
       "      <td>GCAG</td>\n",
       "      <td>1176</td>\n",
       "      <td>638</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>5.425000e+01</td>\n",
       "      <td>1862</td>\n",
       "      <td>X-155239824-ACAA-GCAG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6977 rows  12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     variant_type CHROM        POS               ref_allele  \\\n",
       "0     Subtitution     1     931393                        G   \n",
       "1     Subtitution     1     935222                        C   \n",
       "2     Subtitution     1    1277533                        T   \n",
       "3     Subtitution     1    1284490                        G   \n",
       "4     Subtitution     1    1571850                        G   \n",
       "5     Subtitution     1    1572579                        A   \n",
       "6     Subtitution     1    1575616                        T   \n",
       "7     Subtitution     1    1575784                        C   \n",
       "8     Subtitution     1    1577180                        C   \n",
       "9     Subtitution     1    1635004                        T   \n",
       "10    Subtitution     1    1635749                        C   \n",
       "11    Subtitution     1    1636044                        G   \n",
       "12    Subtitution     1    1638994                        C   \n",
       "13    Subtitution     1    1647778                        C   \n",
       "14    Subtitution     1    1647814                        T   \n",
       "15    Subtitution     1    1647871                        T   \n",
       "16    Subtitution     1    1650787                        T   \n",
       "17    Subtitution     1    1650797                    ATTTT   \n",
       "18    Subtitution     1    1650807                        T   \n",
       "19    Subtitution     1    1650832                        A   \n",
       "20    Subtitution     1    1650845                        G   \n",
       "21    Subtitution     1    1653004                        T   \n",
       "22    Subtitution     1    1653028                        C   \n",
       "23       Deletion     1    1654129                TAAAAAAAT   \n",
       "24    Subtitution     1    2160735                        T   \n",
       "25    Subtitution     1    2164935                        G   \n",
       "26    Subtitution     1    2488153                        A   \n",
       "27    Subtitution     1    2489247                        C   \n",
       "28    Subtitution     1    2985885                        C   \n",
       "29    Subtitution     1    3301721                        C   \n",
       "...           ...   ...        ...                      ...   \n",
       "6947    Insertion     X  135960146                       GA   \n",
       "6948  Subtitution     X  135960230                        C   \n",
       "6949     Deletion     X  135960247              TAAAAAAAAAG   \n",
       "6950     Deletion     X  135961585                    ATGTT   \n",
       "6951     Deletion     X  147582595           CCCGCCGCCGCCGC   \n",
       "6952  Subtitution     X  147919280                        T   \n",
       "6953  Subtitution     X  148035200                        G   \n",
       "6954     Deletion     X  148048333                 CAAAAAAG   \n",
       "6955  Subtitution     X  148055144                        A   \n",
       "6956     Deletion     X  149639632     TGCAGCAGCAGCAGCAGCAA   \n",
       "6957     Deletion     X  149639651           ACAGCAGCAGCAGC   \n",
       "6958  Subtitution     X  149653540                        G   \n",
       "6959  Subtitution     X  149653616                        G   \n",
       "6960  Subtitution     X  153278829                        G   \n",
       "6961  Subtitution     X  153283470                        C   \n",
       "6962  Subtitution     X  153284192                        A   \n",
       "6963  Subtitution     X  153284483                        G   \n",
       "6964  Subtitution     X  153994596                        G   \n",
       "6965     Deletion     X  154005088  CAAGAAGAAGAAGAAGAAGAAGA   \n",
       "6966  Subtitution     X  154294900                        C   \n",
       "6967  Subtitution     X  154998442                        G   \n",
       "6968  Subtitution     X  155003886                        G   \n",
       "6969  Subtitution     X  155004280                        A   \n",
       "6970  Subtitution     X  155007278                        G   \n",
       "6971  Subtitution     X  155009661                        A   \n",
       "6972  Subtitution     X  155011829                        T   \n",
       "6973  Subtitution     X  155011926                        T   \n",
       "6974  Subtitution     X  155233098                        T   \n",
       "6975  Subtitution     X  155239499                        G   \n",
       "6976  Subtitution     X  155239824                     ACAA   \n",
       "\n",
       "            variant_allele  ref_allele_count variant_allele_count  variant_AF  \\\n",
       "0                        T              4029                   95    0.000000   \n",
       "1                        A               480                  652    0.666667   \n",
       "2                        C                 0                  786    1.000000   \n",
       "3                        A                 0                  228    1.000000   \n",
       "4                        A              3961                   94    0.000000   \n",
       "5                        G              3430                   26    0.000000   \n",
       "6                        C                 0                 1118    1.000000   \n",
       "7                        T              3206                  266    0.333333   \n",
       "8                        T                40                   50    0.666667   \n",
       "9                        C                 0                 4076    1.000000   \n",
       "10                       A                 2                 3060    1.000000   \n",
       "11                       A                 0                 3702    1.000000   \n",
       "12                       T               696                  454    0.333333   \n",
       "13                       G              4492                  626    0.333333   \n",
       "14                       C              4958                  632    0.333333   \n",
       "15                       C              4506                  640    0.333333   \n",
       "16                       C              2512                  124    0.000000   \n",
       "17                   GTTTC              2560                   74    0.000000   \n",
       "18                       C               278                 2532    0.666667   \n",
       "19                       G              2874                  378    0.333333   \n",
       "20                       A               376                 2618    0.666667   \n",
       "21                       C                 0                 2376    1.000000   \n",
       "22                       T                 0                 2722    1.000000   \n",
       "23                TAAAAAAT              2116                   80    0.000000   \n",
       "24                       A              2061                   50    0.000000   \n",
       "25                       T              1950                 1818    0.333333   \n",
       "26                       G              1002                  998    0.333333   \n",
       "27                       G              1593                   41    0.000000   \n",
       "28                       G               108                   74    0.333333   \n",
       "29                       T               646                  686    0.666667   \n",
       "...                    ...               ...                  ...         ...   \n",
       "6947                  GAAA              3836                  390    0.333333   \n",
       "6948                     T              2624                  118    0.000000   \n",
       "6949            TAAAAAAAAG              1780                   34    0.000000   \n",
       "6950                  ATTT                 0                  252    0.333333   \n",
       "6951           CCCGCCGCCGC               800                   26    0.000000   \n",
       "6952                     G              1050                 1046    0.333333   \n",
       "6953                     A               828                  586    0.333333   \n",
       "6954               CAAAAAG              4740                   24    0.000000   \n",
       "6955                     G                 0                 2194    1.000000   \n",
       "6956     TGCAGCAGCAGCAGCAA              2654                   72    0.000000   \n",
       "6957           ACAGCAGCAGC              2854                   18    0.000000   \n",
       "6958                     A              1190                  862    0.333333   \n",
       "6959                     T              2198                 1630    0.333333   \n",
       "6960                     A               650                  590    0.333333   \n",
       "6961                     T              2854                   20    0.000000   \n",
       "6962                     G              1110                 1068    0.333333   \n",
       "6963                     A               606                  652    0.666667   \n",
       "6964                     T                 2                 3508    1.000000   \n",
       "6965  CAAGAAGAAGAAGAAGAAGA              1124                   76    0.000000   \n",
       "6966                     A              1778                 1512    0.333333   \n",
       "6967                     C              1438                 1210    0.333333   \n",
       "6968                     A              3030                  830    0.333333   \n",
       "6969                     G              2252                 2212    0.333333   \n",
       "6970                     A                 0                  520    1.000000   \n",
       "6971                     C                 2                 4028    1.000000   \n",
       "6972                     G                 6                 1556    1.000000   \n",
       "6973                     C                 4                 2798    1.000000   \n",
       "6974                     C               522                  460    0.333333   \n",
       "6975                     A               442                   46    0.333333   \n",
       "6976                  GCAG              1176                  638    0.333333   \n",
       "\n",
       "      variant_vs_ref%    DP  \\\n",
       "0        2.360000e+00  4124   \n",
       "1        1.358300e+02  1134   \n",
       "2                 inf   786   \n",
       "3                 inf   228   \n",
       "4        2.370000e+00  4055   \n",
       "5        7.600000e-01  3456   \n",
       "6                 inf  1118   \n",
       "7        8.300000e+00  3476   \n",
       "8        1.250000e+02    90   \n",
       "9                 inf  4076   \n",
       "10       1.530000e+05  3062   \n",
       "11                inf  3702   \n",
       "12       6.523000e+01  1150   \n",
       "13       1.394000e+01  5118   \n",
       "14       1.275000e+01  5590   \n",
       "15       1.420000e+01  5146   \n",
       "16       4.940000e+00  2636   \n",
       "17       2.890000e+00  2636   \n",
       "18       9.107900e+02  2810   \n",
       "19       1.315000e+01  3252   \n",
       "20       6.962800e+02  2994   \n",
       "21                inf  2376   \n",
       "22                inf  2722   \n",
       "23       3.780000e+00  2198   \n",
       "24       2.430000e+00  2111   \n",
       "25       9.323000e+01  3772   \n",
       "26       9.960000e+01  2000   \n",
       "27       2.570000e+00  1634   \n",
       "28       6.852000e+01   184   \n",
       "29       1.061900e+02  1332   \n",
       "...               ...   ...   \n",
       "6947     1.017000e+01  4228   \n",
       "6948     4.500000e+00  2742   \n",
       "6949     1.910000e+00  1860   \n",
       "6950              inf  1306   \n",
       "6951     3.250000e+00   848   \n",
       "6952     9.962000e+01  2096   \n",
       "6953     7.077000e+01  1416   \n",
       "6954     5.100000e-01  4784   \n",
       "6955              inf  2194   \n",
       "6956     2.710000e+00  2772   \n",
       "6957     6.300000e-01  2894   \n",
       "6958     7.244000e+01  2052   \n",
       "6959     7.416000e+01  3832   \n",
       "6960     9.077000e+01  1240   \n",
       "6961     7.000000e-01  2874   \n",
       "6962     9.622000e+01  2178   \n",
       "6963     1.075900e+02  1258   \n",
       "6964     1.754000e+05  3512   \n",
       "6965     6.760000e+00  1206   \n",
       "6966     8.504000e+01  3296   \n",
       "6967     8.414000e+01  2648   \n",
       "6968     2.739000e+01  3860   \n",
       "6969     9.822000e+01  4464   \n",
       "6970              inf   520   \n",
       "6971     2.014000e+05  4030   \n",
       "6972     2.593333e+04  1562   \n",
       "6973     6.995000e+04  2802   \n",
       "6974     8.812000e+01   982   \n",
       "6975     1.041000e+01   488   \n",
       "6976     5.425000e+01  1862   \n",
       "\n",
       "                                             Query_name   AF_ExAC  \n",
       "0                                          1-931393-G-T       NaN  \n",
       "1                                          1-935222-C-A  0.661111  \n",
       "2                                         1-1277533-T-C  0.997571  \n",
       "3                                         1-1284490-G-A  0.897299  \n",
       "4                                         1-1571850-G-A  0.000011  \n",
       "5                                         1-1572579-A-G  0.002177  \n",
       "6                                         1-1575616-T-C  0.843984  \n",
       "7                                         1-1575784-C-T  0.094886  \n",
       "8                                         1-1577180-C-T  0.488372  \n",
       "9                                         1-1635004-T-C  0.580487  \n",
       "10                                        1-1635749-C-A  0.471720  \n",
       "11                                        1-1636044-G-A  0.564420  \n",
       "12                                        1-1638994-C-T  0.164064  \n",
       "13                                        1-1647778-C-G  0.500670  \n",
       "14                                        1-1647814-T-C  0.501101  \n",
       "15                                        1-1647871-T-C  0.500961  \n",
       "16                                        1-1650787-T-C  0.501135  \n",
       "17                                1-1650797-ATTTT-GTTTC       NaN  \n",
       "18                                        1-1650807-T-C  0.478368  \n",
       "19                                        1-1650832-A-G  0.501110  \n",
       "20                                        1-1650845-G-A  0.478357  \n",
       "21                                        1-1653004-T-C  0.848546  \n",
       "22                                        1-1653028-C-T  0.854939  \n",
       "23                         1-1654129-TAAAAAAAT-TAAAAAAT       NaN  \n",
       "24                                        1-2160735-T-A       NaN  \n",
       "25                                        1-2164935-G-T       NaN  \n",
       "26                                        1-2488153-A-G  0.544314  \n",
       "27                                        1-2489247-C-G       NaN  \n",
       "28                                        1-2985885-C-G  0.460045  \n",
       "29                                        1-3301721-C-T  0.329498  \n",
       "...                                                 ...       ...  \n",
       "6947                                X-135960146-GA-GAAA       NaN  \n",
       "6948                                    X-135960230-C-T  0.134586  \n",
       "6949                 X-135960247-TAAAAAAAAAG-TAAAAAAAAG       NaN  \n",
       "6950                             X-135961585-ATGTT-ATTT       NaN  \n",
       "6951             X-147582595-CCCGCCGCCGCCGC-CCCGCCGCCGC       NaN  \n",
       "6952                                    X-147919280-T-G  0.006129  \n",
       "6953                                    X-148035200-G-A  0.073504  \n",
       "6954                       X-148048333-CAAAAAAG-CAAAAAG       NaN  \n",
       "6955                                    X-148055144-A-G  0.982914  \n",
       "6956  X-149639632-TGCAGCAGCAGCAGCAGCAA-TGCAGCAGCAGCA...       NaN  \n",
       "6957             X-149639651-ACAGCAGCAGCAGC-ACAGCAGCAGC       NaN  \n",
       "6958                                    X-149653540-G-A       NaN  \n",
       "6959                                    X-149653616-G-T       NaN  \n",
       "6960                                    X-153278829-G-A  0.685193  \n",
       "6961                                    X-153283470-C-T  0.000012  \n",
       "6962                                    X-153284192-A-G  0.750428  \n",
       "6963                                    X-153284483-G-A  0.652726  \n",
       "6964                                    X-153994596-G-T  0.997436  \n",
       "6965  X-154005088-CAAGAAGAAGAAGAAGAAGAAGA-CAAGAAGAAG...       NaN  \n",
       "6966                                    X-154294900-C-A       NaN  \n",
       "6967                                    X-154998442-G-C       NaN  \n",
       "6968                                    X-155003886-G-A       NaN  \n",
       "6969                                    X-155004280-A-G  0.214952  \n",
       "6970                                    X-155007278-G-A       NaN  \n",
       "6971                                    X-155009661-A-C       NaN  \n",
       "6972                                    X-155011829-T-G       NaN  \n",
       "6973                                    X-155011926-T-C       NaN  \n",
       "6974                                    X-155233098-T-C  0.448144  \n",
       "6975                                    X-155239499-G-A  0.005948  \n",
       "6976                              X-155239824-ACAA-GCAG       NaN  \n",
       "\n",
       "[6977 rows x 12 columns]"
      ]
     },
     "execution_count": 254,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#give a final look before save it\n",
    "report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the final report as csv\n",
    "export_csv = report.to_csv (r'annoted_VCF.csv', index = None, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
